client/src/planwise/client/providers_set/components/dropdown.cljs

```diff
@@ -1,33 +1,16 @@
 (ns planwise.client.providers-set.components.dropdown
-  (:require [re-frame.core :refer [subscribe dispatch]]
-            [reagent.core :as r]
-            [clojure.string :as string]
+  (:require [re-frame.core :as rf]
             [planwise.client.asdf :as asdf]
-            [planwise.client.utils :as utils]
-            [planwise.client.components.common2 :as common2]
-            [planwise.client.providers-set.db :as db]
-            [planwise.client.ui.rmwc :as m]
-            [re-frame.utils :as c]))
-
-(defn- providers-set-select-component
-  [{:keys [label value options empty-label on-change]}]
-  [m/Select {:label (if (empty? options) empty-label label)
-             :disabled (empty? options)
-             :value (str value)
-             :options options
-             :on-change #(on-change (js/parseInt (-> % .-target .-value)))}])
+            [planwise.client.ui.filter-select :as filter-select]))
 
 (defn providers-set-dropdown-component
-  [{:keys [label value on-change disabled?]}]
-  (let [list      (subscribe [:providers-set/list])
-        options   (subscribe [:providers-set/dropdown-options])
-        component (if (or disabled? (empty? @options))
-                    common2/disabled-input-component
-                    providers-set-select-component)]
+  [attrs]
+  (let [list  (rf/subscribe [:providers-set/list])
+        props (merge {:choices   @(rf/subscribe [:providers-set/dropdown-options])
+                      :label-fn  :label
+                      :render-fn (fn [provider-set] [:div.option-row
+                                               [:span (:label provider-set)]])}
+                     attrs)]
     (when (asdf/should-reload? @list)
```

Comment:
I don't know in which scenarios we should reload the list, but it was needed to _at least_ load the options upon page load, so I kept this here.

Reply:
It's pretty usual to realize there's a providers source missing when you get to that screen so, if I get this right, it is good to have a reload after page load